packages/sign-client/src/controllers/engine.ts

```diff
@@ -576,7 +639,14 @@ export class Engine extends IEngine {
       const hash = hashMessage(JSON.stringify(payload));
       this.client.core.verify.register({ attestationId: hash });
     }
-    const message = await this.client.core.crypto.encode(topic, payload);
+    let message;
+    try {
+      message = await this.client.core.crypto.encode(topic, payload);
+    } catch (error) {
+      await this.cleanup();
+      this.client.logger.error(`sendRequest() -> encode(${topic}) failed`);
```

Comment:
This feels misleading because it seems to show a single function parameter when there are two. Think there's less potential for confusion with e.g.:
```suggestion
      this.client.logger.error(`sendRequest() -> core.crypto.encode() for topic ${topic} failed`);
```

Reply:
Also, we can pass objects to `logger`, so might be more useful to move to an approach like this to get the maximum amount of context:

```
logger.error({message: "sendRequest() -> core.crypto.encode...", topic, payload})
```